### ISSUE TITLE
Cross-region support for Bedrock

### ISSUE THREAD

**Issue by shabie** [langchain-ai/langchain-aws#178]
Aamzon Bedrock now supports cross-region inference making it easier to handle throughput: https://aws.amazon.com/about-aws/whats-new/2024/08/amazon-bedrock-cross-region-inference/

This requires using a model ID with the region prefix as can be seen [here](https://docs.aws.amazon.com/bedrock/latest/userguide/cross-region-inference-support.html):

| Inference profile | Inference profile ID | Regions included |
|-------------------|----------------------|-------------------|
| US Anthropic Claude 3 Haiku | us.anthropic.claude-3-haiku-20240307-v1:0 | US East (N. Virginia) (us-east-1)<br>US West (Oregon) (us-west-2) |
| EU Anthropic Claude 3 Haiku | eu.anthropic.claude-3-haiku-20240307-v1:0 | Europe (Frankfurt) (eu-central-1)<br>Europe (Ireland) (eu-west-1)<br>Europe (Paris) (eu-west-3) |
| US Anthropic Claude 3 Opus | us.anthropic.claude-3-opus-20240229-v1:0 | US East (N. Virginia) (us-east-1)<br>US West (Oregon) (us-west-2) |
| US Anthropic Claude 3 Sonnet | us.anthropic.claude-3-sonnet-20240229-v1:0 | US East (N. Virginia) (us-east-1)<br>US West (Oregon) (us-west-2) |
| EU Anthropic Claude 3 Sonnet | eu.anthropic.claude-3-sonnet-20240229-v1:0 | Europe (Frankfurt) (eu-central-1)<br>Europe (Ireland) (eu-west-1)<br>Europe (Paris) (eu-west-3) |
| US Anthropic Claude 3.5 Sonnet | us.anthropic.claude-3-5-sonnet-20240620-v1:0 | US East (N. Virginia) (us-east-1)<br>US West (Oregon) (us-west-2) |
| EU Anthropic Claude 3.5 Sonnet | eu.anthropic.claude-3-5-sonnet-20240620-v1:0 | Europe (Frankfurt) (eu-central-1)<br>Europe (Ireland) (eu-west-1)<br>Europe (Paris) (eu-west-3) |

This means that this code needs to be modified:

https://github.com/langchain-ai/langchain-aws/blob/d89fcf80c61174c21c88ef8d1cfbf3a55c7d84c5/libs/aws/langchain_aws/llms/bedrock.py#L616-L625

This logic should be updated to something like the following:

```python
def _get_provider(self) -> str:
    if self.provider:
        return self.provider
    if self.model_id.startswith("arn"):
        raise ValueError(
            "Model provider should be supplied when passing a model ARN as "
            "model_id"
        )

    first_element, second_element, *_ = self.model_id.split(".")
    if first_element.lower() in {"eu", "us", "ap", "sa"}:
        return second_element
    return first_element
```

I am happy to make a PR.

**Comment by rsgrewal-aws** [comment]
@3coins  and @shabie  can we please urgently get this fix in - this is needed for the Custom Model import feature of Bedrock as well 

for now will sub-class work ?

**Comment by rsgrewal-aws** [comment]
what is the purpose of this code - 

if self.model_id.startswith("arn"):
        raise ValueError(
            "Model provider should be supplied when passing a model ARN as "
            "model_id"
        )

**Comment by kai-zhu-aws** [comment]
https://docs.aws.amazon.com/bedrock/latest/userguide/cross-region-inference-support.html
Seems only us and eu are supported right now.

**Comment by kai-zhu-aws** [comment]
There is a PR for it https://github.com/langchain-ai/langchain-aws/pull/170/files

**Comment by 3coins** [comment]
https://github.com/langchain-ai/langchain-aws/pull/184 fixes this issue. @shabie Can you take a look, there is a slight modification from your code suggestion to allow some of the test runs to pass.

**Comment by Nilanshrajput** [comment]
Hey if we use custom inference profile we create in bedrock, LLM.with_structured_output doesn't work @3coins  any idea what could be the reason, its able to generate on invoke calls but with_structured_output fails, I am using Claude3.5 from bedrock using custom inference profile specific to our project.

**Comment by IndifferentYk** [comment]
The region code of Asia-Pacific should be apac instead of ap. Please fix the bug.

**Comment by 3coins** [comment]
@IndifferentYk 
Thanks for bringing this up. @michaelnchin is working to submit a fix.

**Comment by michaelnchin** [comment]
@IndifferentYk this is fixed with #278 .
